### PR TITLE
Fix #200 Fix #236

### LINKED ISSUE #200: Remove favorites
Need a way to remove a favorited file from the cached files modal.

### LINKED ISSUE #236: Multi add to Bookmarks
you can add the same file multiply times to the bookmarks list.
you an also add bookmarks to the bookmarks list.

### PR DESCRIPTION
**Pull request by mattben**

### CODE REVIEW

**Comment by mattben** [comment]
@James-Crean um... I broke it

**Comment by James-Crean** [comment]
cd into frontend and then run `npm test`

That will give you the test output so you can look at where things went wrong. 

If you find you need to edit and debug test code, the [https://github.com/UV-CDAT/vcdat/tree/master/frontend/test/mocha#writing-tests](README.md) mentions a few steps that will make things easier. (especially the debugging tests part)

File: frontend/src/js/components/modals/CachedFiles/Tabs/FileTab.jsx [hunk 1]
```diff
@@ -244,34 +245,70 @@ class FileTab extends Component {
...
 
-    handleDrop(event){
+    handleDrop(event, option){
         try {
             let file = JSON.parse(event.dataTransfer.getData('text'));
             let bookmarks = this.state.bookmarkFiles.slice()
-            bookmarks.push(file)
+            let fileInBookmarks = false
+            let fileInBookmarksIndex = -1
+            bookmarks.reduce((prev_val, current_obj) => {
```
Comment (James-Crean) [inline frontend/src/js/components/modals/CachedFiles/Tabs/FileTab.jsx]: May need to check the logic around here. Adding bookmarks doesnt always work and removing often removes the wrong bookmark.

File: .gitignore [hunk 2]
```diff
@@ -19,5 +19,6 @@ backend/vcdat/resources
 build/
 dist/
 *.egg-info
+frontend/package-lock.json
```
[the remark this answers is not in the document]
Comment (mattben) [reply .gitignore]: agh, if there's a reason its good enough for me I'll add it back next week.

File: frontend/src/js/components/modals/CachedFiles/Tabs/FileTab.jsx (hunk 1 above)
Comment (mattben) [reply frontend/src/js/components/modals/CachedFiles/Tabs/FileTab.jsx]: :ballot_box_with_check:

File: frontend/src/js/components/modals/CachedFiles/Tabs/FileTab.jsx [hunk 3]
```diff
@@ -244,34 +245,70 @@ class FileTab extends Component {
...
         } catch (e) {
             console.log(e)
             return;
         }
     }
 
-    handleDragEnd(){
-        this.setState({showBookmarkZone: false})
+    handleDragEnd(event, option){
+        console.log("drag end " + option)
```
[the remark this answers is not in the document]
Comment (mattben) [reply frontend/src/js/components/modals/CachedFiles/Tabs/FileTab.jsx]: :ballot_box_with_check:

File: frontend/src/js/components/modals/CachedFiles/Tabs/FileTab.jsx [hunk 4]
```diff
@@ -244,34 +245,70 @@ class FileTab extends Component {
...
+        if(option == 'add'){
+            this.setState({showBookmarkZone: false})
+        }
+        else if(option == 'remove'){
+            this.setState({showRemoveBookmarkZone: false})
+        }
+        else{
+            console.log('Error')
```
[the remark this answers is not in the document]
Comment (mattben) [reply frontend/src/js/components/modals/CachedFiles/Tabs/FileTab.jsx]: :ballot_box_with_check:

**Comment by mattben** [comment]
@James-Crean (@doutriaux1) I addressed all the requested changes, and corrected the test. I added a new parameter, that the test was not passing. Once this passes it should be safe to merge.

**Pull request closed without merging**